lms/services/blackboard_api/client.py

```diff
@@ -59,6 +61,24 @@ def list_files(self, course_id, folder_id=None):
             if not path:
                 break
 
+        # Notify that we've found some files
+        self._request.registry.notify(
+            FilesDiscoveredEvent(
+                request=self._request,
+                values=[
+                    {
+                        "type": "blackboard_file",
+                        "course_id": course_id,
+                        "lms_id": file["id"],
+                        "name": file["name"],
+                        "size": file["size"],
+                        "parent_lms_id": file["parentId"],
+                    }
+                    for file in results
+                ],
+            )
+        )
```

Comment:
This is not covered by any assert on the tests, but it seems that the canvas counterpart is doing the same so I'm assuming there's a reason for it .

Reply:
Added a separate tests that just asserts that the event is indeed emitted, once that happens there's good coverage for the consequences of that :+1: